src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -277,6 +277,92 @@ module Base = struct
     in
     Boolean.Assert.any [Boolean.not is_user_command; verifies]
 
+  let check_timing ~account ~txn_amount ~txn_global_slot =
+    (* calculations should track Transaction_logic.validate_timing *)
+    let open Account.Poly in
+    let open Account.Timing.As_record in
+    let { is_timed
+        ; initial_minimum_balance
+        ; cliff_time
+        ; vesting_period
+        ; vesting_increment } =
+      account.timing
+    in
+    let%bind before_or_at_cliff =
+      Global_slot.Checked.(txn_global_slot <= cliff_time)
+    in
+    let int_of_field field =
+      Snarky_integer.Integer.constant ~m
+        (Bigint.of_field field |> Bigint.to_bignum_bigint)
+    in
+    let zero_int = int_of_field Field.zero in
+    let balance_to_int balance =
+      Snarky_integer.Integer.of_bits ~m @@ Balance.var_to_bits balance
+    in
+    let txn_amount_int =
+      Snarky_integer.Integer.of_bits ~m @@ Amount.var_to_bits txn_amount
+    in
+    let balance_int = balance_to_int account.balance in
+    (* fail early if insufficient balance *)
+    let%bind () =
+      with_label
+        (sprintf "%s: check balance for transaction" __LOC__)
+        (Boolean.Assert.is_true
+           (Snarky_integer.Integer.gte ~m balance_int txn_amount_int))
+    in
+    let curr_min_balance =
```

Comment:
@mrmr1993 Is it possible to mix the monadic and non-monadic APIs in this way?

Reply:
No, this will most likely break. Good catch!

@psteckler you can wrap this in a [`make_checked`](https://github.com/o1-labs/snarky/blob/dde0ac38cf7c16e7f116cbd161a0a3ca2669f247/src/snark_intf.ml#L2241) call to lower this into the monadic API.